### PR TITLE
DDF-2520: Fixed MetacardImpl copy Constructor

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
Fixes the MetacardImpl copy constructor from adding null attributes
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@mcalcote @Lambeaux 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Core APIs](https://github.com/orgs/codice/teams/core-apis)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@lessarderic
@pklinef

 - The copy constructor will no longer copy null attributes

### CODE REVIEW

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java [hunk 1]
```diff
@@ -158,7 +158,11 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
         }
         map = new HashMap<>();
         for (AttributeDescriptor attribute : metacard.getMetacardType().getAttributeDescriptors()) {
-            map.put(attribute.getName(), metacard.getAttribute(attribute.getName()));
+            Attribute metacardAttribute = metacard.getAttribute(attribute.getName());
```
Comment (lessarderic) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: Should we add a note in the comment regarding the fact that `null` attributes and attribute values will _not_ be copied?

Comment (lessarderic) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: We should also add or update the unit tests for this change.

Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: Agreed. Will do.

Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: @lessarderic Reading the documentation it looks like it is already actually saying that.

> ...but will create a new metacard and clone any attributes defined by the {@link MetacardType} that exist on the given {@link Metacard}.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java [hunk 2]
```diff
@@ -158,7 +158,11 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
         }
         map = new HashMap<>();
         for (AttributeDescriptor attribute : metacard.getMetacardType().getAttributeDescriptors()) {
```
Comment (Lambeaux) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: Not mandatory but renaming this to `descriptor` instead of `attribute` would really help with clarity. We throw the word _"attribute"_ around so much without specificity that it's hard to follow precisely which component we're referring to. 

Not part of your change-set so completely optional, I would say.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java (hunk 1 above)
Comment (lessarderic) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: I guess it depends on your definition of _exist_. It could mean the metacard attribute it `null`, or the attribute _value_ is `null`, or both, which isn't very clear from the comment imho. But it could just be me too :-)

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java]: Agreed. Updated.

**Comment by shaundmorris** [comment]
@rzwiefel is there a reason we haven't merged this yet? Also should it be ported to 2.9.x?

**Comment by rzwiefel** [comment]
@shaundmorris Probably need a hero build of it to be safe, and not sure about backport. I'd say this is actually an improvement (even though my title says fixed 😞 ) Nothing is broken its just ... making the behavior more predictable.

**Comment by rzwiefel** [comment]
Attempted Hero, though had erroneous timeouts.. rerunning.

**Comment by rzwiefel** [comment]
DDF Hero build passed
